### ISSUE TITLE
QOpenGLWidget: Failed to create context: UI rendering does not work on Linux when run over network

### ISSUE THREAD

**Issue by jdesbonnet** [colmap/colmap#1201]
When GUI is run on a headless Ubuntu 18.04 when logged over a network (ssh with -X), "colmap gui" opens a window but it is a black rectangle.  However by clicking near the top of window the menus can be accessed but otherwise is generally unusable.  The remote computer has  2 x Titan RTX GPUs and colmap works ok in non-UI mode.

On the command line after "colmap gui" the following info messages appear:

```
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
composeAndFlush: makeCurrent() failed
```

To reproduce:

On clean Ubuntu 18.04 build according to colmap documentation. Log in from another computer using ssh -X (to enable remote X-Window). Run "colmap gui"

colmap GUI builds and runs as expected on my local Ubuntu 18.04 which has a light weight Nvidia GPU and monitor.

**Comment by zehongs** [comment]
Hi, have you found a solution?

**Comment by mariapatni** [comment]
I am having this issue as well, can you help? 

Edit: For anyone else having this issue, look at how I resolved!

In my case, I was remote sshing into a 

Ubuntu 20.04 LTS system
NVIDIA GeForce RTX 2080 Ti GPU 
Driver Version: 570.169
CUDA Version: 12.8

My original goal was to run the colmap GUI from a remote ssh into the device.
When I initially tried to run colmap GUI, it returns

```
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
composeAndFlush: makeCurrent() failed
```

When I initially tried this command:
`$ glxinfo | grep -E "OpenGL vendor|OpenGL renderer|OpenGL version"
`
It returns:
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: llvmpipe (LLVM 12.0.0, 256 bits)
OpenGL version string: 3.1 Mesa 21.2.6

However, it should return
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
OpenGL version string: 4.6.0 NVIDIA 570.169

Open GL needs to use the NVIDIA renderer, not the the Mesa/X one. If you reinstall your display driver (I got mine from [here](https://www.nvidia.com/en-us/drivers/details/247720/)) it should resolve the issue.



TLDR:

If glxinfo says llvmpipe or Mesa and you're using an NVIDIA GPU:

    1. Check for missing /usr/lib/.../nvidia/.../libGL.so.1

     2. If missing, your OpenGL user-space libs are broken

     3. Redownload and run the .run display driver installer from NVIDIA, reboot, and verify
